src/repo.rs

```diff
@@ -31,6 +32,20 @@ fn ensure_if_cros_dir(path: &str) -> Result<()> {
     }
 }
 
+fn find_cros_dir_from_cwd() -> Result<String> {
```

Comment:
hmm, something like this works for me:
```
+fn find_cros_dir_from_cwd() -> Result<String> {
+    let mut path = env::current_dir()?;
+    let mut dir = path.to_string_lossy().to_string();
+
+    while !is_cros_dir(&dir) {
+        match path.parent() {
+            Some(p) => path = p.to_path_buf(),
+            None => return Err(anyhow!("Failed to find Cros SDK dir")),
+        }
+        dir = path.to_string_lossy().to_string();
+    }
+
+    Ok(dir)
+}
+
```

I wonder why comparing to None didn't work for you. Perhaps did you expect `path` to be updated when you do  `Some(path) = ...`?